src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -214,26 +214,19 @@ public async Task GetAsync(ArrayBuilder<DiagnosticData> list, CancellationToken
 
                         if (includeSyntax || includeSemantic)
                         {
-                            var state = stateSet.GetOrCreateActiveFileState(_document.Id);
-
+                            stateSet.AddActiveDocument(_document.Id);
                             if (includeSyntax)
                             {
-                                var existingData = state.GetAnalysisData(AnalysisKind.Syntax);
-                                if (!await TryAddCachedDocumentDiagnosticsAsync(stateSet.Analyzer, AnalysisKind.Syntax, existingData, list, cancellationToken).ConfigureAwait(false))
-                                    syntaxAnalyzers.Add(new AnalyzerWithState(stateSet.Analyzer, stateSet.IsHostAnalyzer, state, existingData));
+                                syntaxAnalyzers.Add(new AnalyzerWithState(stateSet.Analyzer, stateSet.IsHostAnalyzer));
```

Comment:
existingData would always be the default value.  which would cause TryAddCachedDocumentDiagnosticsAsync to always fail.